### PR TITLE
Fix range.moveToRangeOf when range is a backward range;

### PR DESCRIPTION
**Pull request by zhujinxuan**
@CameronAckermanSEL   Can you have a test whether unwrapBlock is good after this PR?

It should fix https://github.com/ianstormtaylor/slate/issues/1500
The problem is that moveToRangeOf does not consider the case `Range.isBackward`.  I think there can be two options:
1. `moveToRangeOf` always return with `isBackward: false`
2. `moveToRangeOf` keep `isBackward`, but switch anchor and focus if necessary.

In this PR, I choose the second solution

### CODE REVIEW

**Comment by CameronAckermanSEL** [comment]
I'm really confident that this will fix the defect. Nice job @zhujinxuan !

**Comment by ianstormtaylor** [comment]
Hey @zhujinxuan I'm not totally sure I understand, but I think that (1) makes more sense a solution? That way you know what to expect no matter what when you call `moveToRangeOf`?

**Comment by ererik** [comment]
Here's a test that covers the fail condition:

`packages\ slate\test\changes\at-current-range\unwrap-block\middle-child-block-reversed.js`

```
/** @jsx h */

import h from '../../../helpers/h'

export default function (change) {
  change.unwrapBlock('quote')
}

export const input = (
  <value>
    <document>
      <quote>
        <paragraph>
          one
        </paragraph>
        <paragraph>
          t<focus />w<anchor />o
        </paragraph>
        <paragraph>
          three
        </paragraph>
      </quote>
    </document>
  </value>
)

export const output = (
  <value>
    <document>
      <quote>
        <paragraph>
          one
        </paragraph>
      </quote>
      <paragraph>
        t<focus />w<anchor />o
      </paragraph>
      <quote>
        <paragraph>
            three
        </paragraph>
      </quote>
    </document>
  </value>
)

```

**Comment by zhujinxuan** [comment]
@ianstormtaylor The problem is caused by the following code and alike codes in `by-key.js`
```
  const { value } = change
  const { document, selection } = value
  ....
  const range = selection.moveToRangeOf(first, last)
```

The problem is that, when the selection is backward,  the original `moveToRangeOf` cannot switch the anchor and focus wisely.

**Comment by zhujinxuan** [comment]
@ererik Thanks. The test is added

File: packages/slate/src/models/range.js [hunk 1]
```diff
@@ -657,6 +657,10 @@ class Range extends Record(DEFAULTS) {
    */
 
   moveToRangeOf(start, end = start) {
+    if (this.isBackward) {
+      return this.flip().moveAnchorToStartOf(start)
+        .moveFocusToEndOf(end)
+    }
```
Comment (ianstormtaylor) [inline packages/slate/src/models/range.js]: Can you change this to only have a single code path, so we're not duplicating the `moveAnchorToStartOf` stuff? For example:

```
let range = this

if (this.isBackward) {
  range = this.flip()
}

return range
  ...
```

Comment (zhujinxuan) [reply packages/slate/src/models/range.js]: Fixed. thanks

**Comment by ianstormtaylor** [comment]
Thanks @zhujinxuan!

**Pull request merged**